File: src/api/system/about.ts
import { post } from '@/utils/request'

export function get<T>() {
  return post<T>({
    url: '/about',
  })
}

export function getSiteSetting<T>() {
  return post<T>({
    url: '/system/siteSetting/get',
  })
}

export function setSiteSetting<T>(data: System.SiteSettingRequest) {
  return post<T>({
    url: '/system/siteSetting/set',
    data,
  })
}

File: service/router/system/about.go
package system

import (
	"zpanel/api/api_v1"
	"zpanel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitAbout(router *gin.RouterGroup) {
	about := api_v1.ApiGroupApp.ApiSystem.About
	{
		router.POST("about", about.Get)
		router.POST("system/siteSetting/get", middleware.LoginInterceptor, about.GetSiteSetting)
		router.POST("system/siteSetting/set", middleware.LoginInterceptor, middleware.AdminInterceptor, about.SetSiteSetting)
	}
}
